modules/gdnative/gdnative/string.cpp

```diff
@@ -30,57 +30,15 @@
 
 #include "gdnative/string.h"
 
-#include "core/string/string_name.h"
 #include "core/string/ustring.h"
-#include "core/variant/variant.h"
 
-#include <string.h>
+static_assert(sizeof(godot_string) == sizeof(String), "String size mismatch");
+static_assert(sizeof(godot_char_type) == sizeof(char32_t), "char32_t size mismatch");
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-static_assert(sizeof(godot_char16_string) == sizeof(Char16String), "Char16String size mismatch");
-static_assert(sizeof(godot_char_string) == sizeof(CharString), "CharString size mismatch");
```

Comment:
Are you sure that `CharString` and `Char16String` don't need to bound anymore? (I don't know, just raising it for discussion). CC @bruvzg

Reply:
I don't see how having those as types can be useful. AFAIK Godot doesn't deal with those directly in the API, but only with Strings. External code can use their own char types, since there are functions to make a `godot_string` out of those.